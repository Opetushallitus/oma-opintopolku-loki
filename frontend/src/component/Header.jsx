import React from 'react'
import { H1, Lead } from 'ui/typography'
import t from 'util/translate'
import StudentInfo from 'component/generic/widget/StudentInfo'
import NavBar from 'component/generic/widget/NavBar'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const Description = styled(Lead)`
    margin-top: 2.429rem;
    margin-bottom: 2.429rem;
`

const Header = ({ selectedHetu, onSelectHetu }) => (
  <header>
    <H1>
      {t`Tietojeni käyttö`}
    </H1>
    <Description>
      {t`Tällä sivulla voit tarkastella ja hallinnoida antamiasi käyttölupia tietoihisi. Lisäksi näet mitkä toimijat, esim. viranomaiset, ovat käyttäneet tietojasi.`}{'*'}
    </Description>

    <StudentInfo selectedHetu={selectedHetu} onSelectHetu={onSelectHetu}/>
    <NavBar />
  </header>
)
Header.propTypes = ({
  selectedHetu: PropTypes.string,
  onSelectHetu: PropTypes.func
})

export default Header
