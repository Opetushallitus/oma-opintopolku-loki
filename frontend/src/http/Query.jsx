import React from 'react'
import PropTypes from 'prop-types'
import http from './http'

class Query extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      data: null,
      error: null,
      pending: true
    }
  }

  componentDidMount () {
    this.doQuery()
  }

  async doQuery () {
    const { url, method, body } = this.props

    try {
      const { data } = await http.request(url, method, body)
      this.setState({ data, pending: false })
    } catch (error) {
      console.error(error)
      this.setState({ error, pending: false })
    }
  }

  render () {
    return this.props.children(this.state)
  }
}

Query.defaultProps = {
  method: 'get',
  body: {}
}

Query.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string,
  body: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array
  ]),
  children: PropTypes.func.isRequired
}

export default Query
