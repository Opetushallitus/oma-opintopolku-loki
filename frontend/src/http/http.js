import axios from 'axios'

const client = axios.create({
  baseURL: process.env.API_BASE_URL
})

const get = async url => client.get(url)
const post = async (url, data) => client.post(url, data)

const request = async (url, method, data = {}) => {
  switch (method) {
    case 'GET':
    case 'get':
      return get(url)
    case 'POST':
    case 'post':
      return post(url, data)
    default:
      throw new Error(`Unsupported HTTP method: ${method}`)
  }
}

export default {
  get,
  post,
  request
}
