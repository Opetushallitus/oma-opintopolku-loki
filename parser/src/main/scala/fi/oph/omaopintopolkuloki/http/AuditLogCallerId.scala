package fi.oph.omaopintopolkuloki.http

object AuditLogCallerId {
  val code = "1.2.246.562.10.00000000001.oma-opintopolku-loki/parser"
}
