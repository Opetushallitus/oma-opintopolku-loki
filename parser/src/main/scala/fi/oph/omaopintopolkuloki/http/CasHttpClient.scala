package fi.oph.omaopintopolkuloki.http

import fi.oph.omaopintopolkuloki.conf.Configuration
import fi.vm.sade.utils.cas.{CasAuthenticatingClient, CasClient, CasParams}
import org.http4s.client._


object CasHttpClient {

  val sessionCookieName = "JSESSIONID"

  def apply(httpClient: Client, scheme_authority: String, params: CasParams = Params.permission) = {
    CasAuthenticatingClient(
      new CasClient(scheme_authority, httpClient),
      params,
      httpClient,
      Some(AuditLogParserSubsystemCode.code),
      sessionCookieName
    )
  }
}


object Params {
  private lazy val permission_path = "/kayttooikeus-service"

  private lazy val credentials = Configuration.getSecrets

  def permission: CasParams = CasParams(permission_path, credentials.username, credentials.password)
}

