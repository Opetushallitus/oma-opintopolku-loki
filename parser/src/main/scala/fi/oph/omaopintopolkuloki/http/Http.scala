package fi.oph.omaopintopolkuloki.http

import cats.effect.IO
import cats.effect.unsafe.IORuntime
import fi.oph.omaopintopolkuloki.conf.Configuration._
import org.http4s.blaze.client.BlazeClientBuilder
import org.http4s.client.Client
import org.http4s.{Request, Uri}

import scala.concurrent.ExecutionContext.global

trait HttpClient {
  type Decode[ResultType] = (Int, String, Request[IO]) => ResultType

  def get[ResultType](uri: Uri)(decode: Decode[ResultType]): IO[ResultType]
}

object Http {
  implicit private val runtime: IORuntime = cats.effect.unsafe.IORuntime.global

  def apply(useCas: Boolean = false): HttpClient = {
    val client = BlazeClientBuilder[IO](global)
      .withMaxTotalConnections(maxRequestThreads)
      .allocated
      .map(_._1)
      .unsafeRunSync()

    if (useCas) {
      new Http(CasHttpClient(client, baseURI.renderString + "/cas"))
    } else {
      new Http(client)
    }
  }
}

private class Http(client: Client[IO]) extends HttpClient {

  def get[ResultType](uri: Uri)(decode: Decode[ResultType]): IO[ResultType] = {
    processRequest(Request[IO](uri = uri))(decode)
  }

  private def processRequest[ResultType](request: Request[IO])(decoder: (Int, String, Request[IO]) => ResultType): IO[ResultType] = {
    client.run(request).use { response =>
      response.as[String].map(text => decoder(response.status.code, text, request))
    }
  }
}
