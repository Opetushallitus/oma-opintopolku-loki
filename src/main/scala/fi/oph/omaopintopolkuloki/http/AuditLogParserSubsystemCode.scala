package fi.oph.omaopintopolkuloki.http

object AuditLogParserSubsystemCode {
  val code = "auditlog"
}
