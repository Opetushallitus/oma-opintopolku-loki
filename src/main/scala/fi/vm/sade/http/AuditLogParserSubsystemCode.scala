package fi.vm.sade.http

object AuditLogParserSubsystemCode {
  val code = "auditlog"
}
