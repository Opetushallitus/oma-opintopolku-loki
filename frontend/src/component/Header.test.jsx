import React from 'react'
import Header from './Header'
import { create } from 'react-test-renderer'
import http from '../http/http'

describe('Header', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('should be rendered', (done) => {
    http.request = jest.fn(() =>
      Promise.resolve({
        data: {
          etunimet: 'Onni Oskari',
          sukunimi: 'Oppija',
          hetu: '010280-ABC1'
        }
      })
    )

    const renderer = create(<Header/>)

    setTimeout(() => {
      const componentJSON = renderer.toJSON()
      expect(componentJSON).toMatchSnapshot()
      renderer.unmount()

      expect(http.request).toHaveBeenCalledTimes(1)

      done()
    })
  })
})
