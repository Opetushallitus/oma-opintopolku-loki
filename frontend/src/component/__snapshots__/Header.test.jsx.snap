// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be rendered 1`] = `
<header>
  <h1
    className="typography__H1-sc-1yq3b4j-0 bzXKrb"
  >
    Tietojeni käyttö
  </h1>
  <p
    className="Header__Description-sc-19ub9le-0 gqzQDt"
  >
    Tällä sivulla voit tarkastella ja hallinnoida antamiasi käyttölupia tietoihisi. Lisäksi näet mitkä tahot, esim. viranomaiset, ovat katsoneet opintotietojasi.
    *
  </p>
  <h3
    className="StudentInfo__Student-sc-8zvgal-0 hsvIvK"
  >
    <span
      className="StudentInfo__Name-sc-8zvgal-1 fKcnY"
    >
      Onni Oskari Oppija
    </span>
    <span
      className="StudentInfo__Birthday-sc-8zvgal-2 igEQxQ"
    >
       s. 
      1.2.1980
    </span>
  </h3>
  <div
    className="NavBar__Navigation-sc-18mzj0n-0 jHKwkt"
  >
    <a
      className="NavBar__Link-sc-18mzj0n-5 ktidaL"
      href="/koski/omadata/kayttooikeudet"
    >
      <span
        className="NavBar__InactiveTab-sc-18mzj0n-2 dhKxFl"
      >
        Annetut käyttöluvat
      </span>
    </a>
    <span
      className="NavBar__ActiveTab-sc-18mzj0n-3 erWAMW"
    >
      Tietojani käyttäneet toimijat
    </span>
    <span
      className="NavBar__Filler-sc-18mzj0n-4 jTKRzr"
    />
  </div>
</header>
`;
